sentry-android-core/src/main/java/io/sentry/android/core/ANRWatchDog.java

```diff
@@ -22,60 +22,83 @@ final class ANRWatchDog extends Thread {
   private final boolean reportInDebug;
   private final ANRListener anrListener;
   private final MainLooperHandler uiHandler;
+  private final TimeProvider timeProvider;
+  /** the interval in which we check if there's an ANR, in ms */
+  private long pollingIntervalMs;
+
   private final long timeoutIntervalMillis;
   private final @NotNull ILogger logger;
-  private final AtomicLong tick = new AtomicLong(0);
+
+  private volatile long lastKnownActiveUiTimestampMs = 0;
   private final AtomicBoolean reported = new AtomicBoolean(false);
 
   private final @NotNull Context context;
 
   @SuppressWarnings("UnnecessaryLambda")
-  private final Runnable ticker =
-      () -> {
-        tick.set(0);
-        reported.set(false);
-      };
+  private final Runnable ticker;
 
   ANRWatchDog(
       long timeoutIntervalMillis,
       boolean reportInDebug,
       @NotNull ANRListener listener,
       @NotNull ILogger logger,
       final @NotNull Context context) {
-    this(timeoutIntervalMillis, reportInDebug, listener, logger, new MainLooperHandler(), context);
+    this(
+        () -> SystemClock.uptimeMillis(),
+        timeoutIntervalMillis,
+        500,
```

Comment:
dunno if posting to main thread every 500ms is fine, but I guess it is?

Reply:
Yes, the task posted is pretty lightweight as well - so this shouldn't have any impact